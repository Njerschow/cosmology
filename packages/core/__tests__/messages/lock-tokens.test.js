import { osmosis } from '../../src/proto/generated/codecimpl';

it('fromObject', () => {
  const msg = osmosis.lockup.MsgLockTokens.fromObject({
    owner: 'osmo1RedactedAddress',
    duration: {
      seconds: '1209600',
      nanos: 0
    },
    coins: [
      {
        amount: '10236850525923652977',
        denom: 'gamm/pool/3'
      }
    ]
  });
  console.log(JSON.stringify(msg, null, 2));
  expect(msg).toMatchSnapshot();
});
it('fromObject', () => {
  const msg = osmosis.lockup.MsgLockTokens.fromObject({
    owner: 'osmo1RedactedAddress',
    duration: {
      seconds: '1209600',
      nanos: 0
    },
    coins: [
      {
        amount: '10236850525923652977',
        denom: 'gamm/pool/3'
      }
    ]
  });
  console.log(JSON.stringify(msg, null, 2));
});

// it('fromPartial', () => {
//   const msg = osmosis.lockup.MsgLockTokens
//     owner: 'osmo1RedactedAddress',
//     duration: '1209600',
//     coins: [
//       {
//         amount: '10236850525923652977',
//         denom: 'gamm/pool/3'
//       }
//     ]
//   });
//   console.log(JSON.stringify(msg, null, 2));
// });
