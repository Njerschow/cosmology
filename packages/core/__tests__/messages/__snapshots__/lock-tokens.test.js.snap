// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fromObject 1`] = `
Object {
  "coins": Array [
    Object {
      "amount": "10236850525923652977",
      "denom": "gamm/pool/3",
    },
  ],
  "duration": Object {
    "nanos": 0,
    "seconds": "1209600",
  },
  "owner": "osmo1RedactedAddress",
}
`;
